src/gc.rs

```diff
@@ -0,0 +1,76 @@
+use crate::Sodg;
+use anyhow::{Context, Result};
+use std::collections::VecDeque;
+
+impl Sodg {
+    pub(crate) fn collect(&mut self, v: u32) -> Result<()> {
```

Comment:
@UARTman can you please add documentation block for this function. Without it, it's hard to understand what it's for and what it does